plugin/trino-hive/src/main/java/io/trino/plugin/hive/fs/TrinoFileSystemCache.java

```diff
@@ -99,6 +103,7 @@ private synchronized FileSystem getInternal(URI uri, Configuration conf, long un
             FileSystem fileSystem = createFileSystem(uri, conf);
             fileSystemHolder = new FileSystemHolder(fileSystem, privateCredentials);
             map.put(key, fileSystemHolder);
+            STATS.incrementCacheSize();
```

Comment:
I figured it was un-necessary since this code block does not change the size of the `map`. It does a `map.remove(key)` followed by a `map.put(key,..)` so it is simply replacing the value for an element in the `map`.

Reply:
> `map.remove(key)` followed by a `map.put(key,..)` so it is simply replacing the value for an element in the `map`.

correct, if we know that the `key` was in the map. is it always the case?